packages/x-scheduler/src/primitives/use-event-calendar/tests/core.EventCalendarInstance.test.ts

```diff
@@ -0,0 +1,250 @@
+
+const adapter = getAdapter();
+const DEFAULT_PARAMS = { events: [] };
+
+describe('Core - EventCalendarInstance', () => {
+  describe('create', () => {
+    it('initializes default state', () => {
+      const { store } = EventCalendarInstance.create(DEFAULT_PARAMS, adapter);
+
+      expect(store.state.view).to.equal(DEFAULT_VIEW);
+      expect(store.state.views).to.deep.equal(DEFAULT_VIEWS);
+      expect(store.state.resources).to.deep.equal([]);
+      expect(store.state.ampm).to.equal(true);
+      expect(store.state.showCurrentTimeIndicator).to.equal(true);
+      expect(store.state.preferences).to.deep.equal(DEFAULT_PREFERENCES);
+      expect(store.state.preferencesMenuConfig).to.deep.equal(DEFAULT_PREFERENCES_MENU_CONFIG);
+
+      const expectedToday = adapter.startOfDay(adapter.date());
+      expect(store.state.visibleDate).toEqualDateTime(expectedToday);
+    });
+
+    it('keeps provided events array', () => {
+      const events = [
+        buildEvent(
+          '1',
+          'Event 1',
+          adapter.date('2025-08-01T08:00:00Z'),
+          adapter.date('2025-08-01T09:00:00Z'),
+        ),
+        buildEvent(
+          '2',
+          'Event 2',
+          adapter.date('2025-09-01T08:00:00Z'),
+          adapter.date('2025-09-01T09:00:00Z'),
+        ),
+      ];
+
+      const { store } = EventCalendarInstance.create({ events }, adapter);
+
+      expect(store.state.events).to.have.length(2);
+      expect(store.state.events[0].title).to.equal('Event 1');
+      expect(store.state.events[1].title).to.equal('Event 2');
+      expect(store.state.events).to.equal(events);
+    });
+  });
+
+  describe('updater', () => {
+    it('syncs partial state from new parameters (events/resources/views/flags/ampm/indicator)', () => {
+      const { store, updater } = EventCalendarInstance.create(DEFAULT_PARAMS, adapter);
+
+      const newParams = {
+        events: [
+          buildEvent(
+            '1',
+            'Test Event',
+            adapter.date('2025-07-01T10:00:00Z'),
+            adapter.date('2025-07-01T11:00:00Z'),
+          ),
+        ],
+        resources: [
+          { id: 'r1', name: 'Resource 1' },
+          { id: 'r2', name: 'Resource 2' },
+        ],
+        views: ['day', 'week'] as CalendarView[],
+        areEventsDraggable: true,
+        areEventsResizable: true,
+        ampm: false,
+        showCurrentTimeIndicator: false,
+      };
+
+      updater(newParams, adapter);
+      expect(getIds(store.state.events)).to.deep.equal(['1']);
+      expect(getIds(store.state.resources)).to.deep.equal(['r1', 'r2']);
+      expect(store.state.views).to.deep.equal(['day', 'week']);
+      expect(store.state.areEventsDraggable).to.equal(true);
+      expect(store.state.areEventsResizable).to.equal(true);
+      expect(store.state.ampm).to.equal(false);
+      expect(store.state.showCurrentTimeIndicator).to.equal(false);
+    });
+
+    it('respects controlled `view` (updates to new value)', () => {
```

Comment:
This test could live in `view.EventCalendar.test.tsx` (same for other features)
But as you want, both are fine for me